frontend/src/components/core/Block/ElementNodeRenderer.tsx

```diff
@@ -214,7 +218,9 @@ const RawElementNodeRenderer = (
       return <Audio width={width} element={node.element.audio as AudioProto} />
 
     case "balloons":
-      return <Balloons scriptRunId={props.scriptRunId} />
+      return (
+        <Balloons scriptRunId={props.scriptRunId} isVisible={!props.isStale} />
```

Comment:
Have we considered putting the conditional in here for the `isStale` prop to return the React Fragment here? I would think it would do the following:

1. Make the components cleaner and not having to worry about visibility
2. Reduce pollution in the React Tree (why render balloons if in the end, it renders a fragment).
3. Lead way for a generalized approach for these type of components with a "one-time" visibility

If we want to think about (3). We could just create a wrapper function for this.

```javascript
function hideIfStale(isStale, component) {
  return isStale ? <></> : component 
}


// ...
case "balloons":
      return hideIfStale(props.isStale, <Balloons scriptRunId={props.scriptRunId} />)
//...
case "snow":
      return hideIfStale(props.isStale, <Snow scriptRunId={props.scriptRunId} />)
```

I think it would be a cleaner way to test `hideIfStale` and worry less about testing the components themselves.

Reply:
Really like the generalized wrapper function approach w/ `isStale` to simplify this implementation/testing!

Another plus is it avoids passing a prop `isVisible` to conditionally render the entire component, which feels a little weird.

Reply:
Great idea. I am already starting on its implementation.

Reply:
Updated the PR.